src/Compiler/CompilerUtils.cpp

```diff
@@ -611,31 +611,39 @@ void registerDialects(mlir::MLIRContext &context) {
 }
 
 // Return 0 on success, error number on failure.
-int processInputFile(std::string inputFilename, mlir::MLIRContext &context,
+int processInputFile(StringRef inputFilename, mlir::MLIRContext &context,
     mlir::OwningOpRef<ModuleOp> &module, std::string *errorMessage) {
-  // Decide if the input file is an ONNX model or a model specified
-  // in MLIR. The extension of the file is the decider.
-  std::string extension =
-      inputFilename.substr(inputFilename.find_last_of(".") + 1);
-  bool inputIsONNX = (extension == "onnx");
-  bool inputIsMLIR = (extension == "mlir");
-
-  if (!inputIsONNX && !inputIsMLIR) {
-    *errorMessage = "Invalid input file '" + inputFilename +
-                    "': Either an ONNX model (.onnx), or an MLIR file (.mlir) "
+  // Decide if the input file is an ONNX model (either ONNX protobuf or JSON) or
+  // a model specified in MLIR. The extension of the file is the decider or, if
+  // the file is "-", then it's STDIN and we treat it as JSON.
+  bool inputIsONNX = false;
+  bool inputIsJSON = false;
+  bool inputIsMLIR = false;
+  if (inputFilename == "-") {
```

Comment:
Is there a reason that inputs from stdin must be JSON format? That feels a bit arbitrary. Wonder if we don't want an option
`-stdin-format=json|mlir|onnx`
and have onnx being the default.

Reply:
it was just to facilitate chaining json parse lit tests with sed, but now that onnx-mlir ignores // comment lines and I removed sed from the scripts there is no longer any need to read from stdin, so I removed the read-from-stdin logic again

Reply:
oh, I like that, thanks